lib/src/streams/stream_handler.dart

```diff
@@ -384,9 +384,9 @@ class StreamHandler extends Object with TerminatableMixin, ClosableMixin {
     }
 
     if (msg is DataStreamMessage) {
-      _sendData(stream, msg.bytes, endStream: false);
+      _sendData(stream, msg.bytes, endStream: msg.endStream);
```

Comment:
Maybe it would make sense then to do the same for incoming `DataStreamMessage` / `HeadersStreamMessage`?

This would e.g. allow the gRPC client see already in the last message that it's the last one (and doesn't need to wait for the `Stream incomingMessages` to be closed.